homeassistant/components/lektrico/sensor.py

```diff
@@ -0,0 +1,291 @@
+        native_unit_of_measurement=POWER_KILO_WATT,
+    ),
+    SessionEnergySensorEntityDescription(
+        key="session_energy",
+        name="Session Energy",
+        device_class=SensorDeviceClass.ENERGY,
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+    ),
+    TemperatureSensorEntityDescription(
+        key="temperature",
+        name="Temperature",
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=TEMP_CELSIUS,
+    ),
+    TotalChargedEnergySensorEntityDescription(
+        key="total_charged_energy",
+        name="Total Charged Energy",
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        device_class=SensorDeviceClass.ENERGY,
+        native_unit_of_measurement=ENERGY_KILO_WATT_HOUR,
+    ),
+    VoltageSensorEntityDescription(
+        key="voltage",
+        name="Voltage",
+        device_class=SensorDeviceClass.VOLTAGE,
+        native_unit_of_measurement=ELECTRIC_POTENTIAL_VOLT,
+    ),
+    InstallCurrentSensorEntityDescription(
+        key="install_current",
+        name="Install Current",
+        device_class=SensorDeviceClass.CURRENT,
+        native_unit_of_measurement=ELECTRIC_CURRENT_AMPERE,
+    ),
+    DynamicCurrentSensorEntityDescription(
+        key="dynamic_current",
+        name="Dynamic Current",
+        device_class=SensorDeviceClass.CURRENT,
+        native_unit_of_measurement=ELECTRIC_CURRENT_AMPERE,
+    ),
+    LedMaxBrightnessSensorEntityDescription(
+        key="led_max_brightness",
+        name="Led Brightness",
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.ILLUMINANCE,
+        native_unit_of_measurement=PERCENTAGE,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Lektrico charger based on a config entry."""
+    _lektrico_device: LektricoDeviceDataUpdateCoordinator = hass.data[DOMAIN][
+        entry.entry_id
+    ]
+
+    sensors = [
+        LektricoSensor(
+            sensor_desc,
+            _lektrico_device,
+            entry.data[CONF_FRIENDLY_NAME],
+        )
+        for sensor_desc in SENSORS
+    ]
+
+    async_add_entities(sensors, False)
+
+
+class LektricoSensor(CoordinatorEntity, SensorEntity):
+    """The entity class for Lektrico charging stations sensors."""
+
+    entity_description: LektricoSensorEntityDescription
+
+    def __init__(
+        self,
+        description: LektricoSensorEntityDescription,
+        _lektrico_device: LektricoDeviceDataUpdateCoordinator,
+        friendly_name: str,
+    ) -> None:
+        """Initialize Lektrico charger."""
+        super().__init__(_lektrico_device)
+        self.friendly_name = friendly_name
+        self.serial_number = _lektrico_device.serial_number
+        self.board_revision = _lektrico_device.board_revision
```

Comment:
These are no longer needed when moving the DeviceInfo to the constructor.

Reply:
I removed them, thanks.